components/whitelist.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormControl,
+  FormMessage,
+} from "@/components/ui/form";
+import { useForm } from "react-hook-form";
+import { useState } from "react";
+
+export const whitelistSchema = z.object({
+  email: z.string().email({ message: "Email address is invalid." }),
+});
+
+export type WhitelistFormData = z.infer<typeof whitelistSchema>;
+
+export default function WhitelistPage() {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const form = useForm<WhitelistFormData>({
+    resolver: zodResolver(whitelistSchema),
+    defaultValues: {
+      email: "",
+    },
+  });
+
+  const handleEmailSubmit = async (data: WhitelistFormData) => {
+    // TODO: endpoint on server /whitelist
+    if (isSubmitting) return;
+    setIsSubmitting(true);
+    setErrorMessage(null);
+    try {
+      const formData = new FormData();
+      formData.append("email", data.email);
+      // const response = await whitelist(formData);
+      // if (response?.error) {
+      //   setErrorMessage(response.error);
+      // }
+      // else {
+      //   form.reset();
+      // }
```

Comment:
on success response, we want to redirect to /pricing page `router.push("/pricing")`